### PR TITLE
facet search

### PR DESCRIPTION
**Pull request by snathanvj**
Output format:
```
[{
	"treatment": [{
		"_id": "Invasive ventilation",
		"count": 4
	}, {
		"_id": "Inhaled Nitric Oxide",
		"count": 3
	}, {
		"_id": "Any Oxygen therapy",
		"count": 3
	}, {
		"_id": "Chloroquine/hydroxychloroquine",
		"count": 1
	}, {
		"_id": "Ventilation",
		"count": 1
	}],
	"license": [{
		"_id": null,
		"count": 6
	}, {
		"_id": "BSD",
		"count": 2
	}, {
		"_id": "MIT",
		"count": 2
	}, {
		"_id": "Apache",
		"count": 1
	}],
	"territories": [{
		"_id": "UK",
		"count": 3
	}, {
		"_id": "Lapland",
		"count": 2
	}, {
		"_id": "France",
		"count": 2
	}, {
		"_id": "Italy",
		"count": 2
	}, {
		"_id": "Trondheim",
		"count": 1
	}]
}]
```

### CODE REVIEW

File: src/main/java/uk/ac/ebi/biosamples/cohortatlas/service/FacetService.java [hunk 1]
```diff
@@ -2,65 +2,31 @@
...
-        )),
-        new Facet("complications", "Complications", "dataSummary.complications", 5, List.of(
-            new FacetValue("Bronchiolitis", 5),
-            new FacetValue("Cardiac arrest", 5),
-            new FacetValue("Myocardial infarction", 5),
-            new FacetValue("Pneumothorax", 5),
-            new FacetValue("Bacterial pneumonia", 5)
-        ))
-    );
-  }
```
Comment (theisuru) [inline src/main/java/uk/ac/ebi/biosamples/cohortatlas/service/FacetService.java:61]: This is the format we expects. This way we do not need to hard code anything on the client side.

File: src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java [hunk 2]
```diff
@@ -6,6 +6,6 @@
 @Data
 @AllArgsConstructor
 public class FacetValue {
-  private String label;
```
Comment (theisuru) [inline src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java:9]: good to have label or value of something meaningful  than _id.

File: src/main/java/uk/ac/ebi/biosamples/cohortatlas/controller/FacetController.java [hunk 3]
```diff
@@ -21,9 +20,8 @@ public FacetController(FacetService facetService) {
   }
 
   @GetMapping()
-  public ResponseEntity<List<Facet>> getFacets(@RequestParam(required = false) String text,
-                                               @RequestParam(value = "filter", required = false) List<String> filters) {
-    List<Facet> facets = facetService.getFacets(text, filters);
+  public ResponseEntity<List<FacetResult>> getFacets() {
```
Comment (theisuru) [inline src/main/java/uk/ac/ebi/biosamples/cohortatlas/controller/FacetController.java:23]: how would list of facetResults would look like. Is that the expected output?

File: src/main/resources/env/init_db_schema.json [hunk 4]
```diff
@@ -137,6 +140,7 @@
     "provider": {
       "name": "EMC"
     },
+    "license": "BSD",
```
Comment (theisuru) [inline src/main/resources/env/init_db_schema.json:143]: These values are taken from recodid and early_cause cohorts and reflect real values. So we can not add "license" filed in any of these cohorts.

**Review by theisuru — CHANGES_REQUESTED** [review]

File: src/main/java/uk/ac/ebi/biosamples/cohortatlas/service/FacetService.java (hunk 1 above)
Comment (snathanvj) [reply src/main/java/uk/ac/ebi/biosamples/cohortatlas/service/FacetService.java:61]: Added complex filters and updated format

File: src/main/resources/env/init_db_schema.json (hunk 4 above)
Comment (snathanvj) [reply src/main/resources/env/init_db_schema.json:161]: ok, just added for testing , will remove it.

File: src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java (hunk 2 above)
Comment (snathanvj) [reply src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java:9]: _id is default output label for some of the aggragate queries in mongo db. If it is really important I can create a new Bean and convert it.

Comment (theisuru) [reply src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java:9]: I think so to keep the API meaningful. Atleast copying that field to "label" field

Comment (snathanvj) [reply src/main/java/uk/ac/ebi/biosamples/cohortatlas/model/FacetValue.java:9]: `
    {
        "category": "dataTypes",
        "displayName": "Data Types",
        "searchPath": "dataTypes",
        "count": 5,
        "values": [
            {
                "count": 2,
                "label": "phenotypicData"
            },
            {
                "count": 1,
                "label": "genomicData"
            },
            {
                "count": 1,
                "label": "environmentalData"
            },
            {
                "count": 1,
                "label": "biospecimens"
            }
        ]
    },`

**Pull request merged**